samples/boards/stm32/spi_flash/README.rst

```diff
@@ -0,0 +1,42 @@
+.. zephyr:code-sample:: spi-nor
```

Comment:
I must be missing something but why are you duplicating the existing sample (including duplicating it's name/id, which probably causes issues in the doc) instead of just adding the needed overlays in the original one?